instruments/abstract_instruments/cryomagnetics_instrument.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""
+Extends the instrument class, and overwrites `Instrument.open_serial`, in order
+to use the `CryomagneticsSerialCommunicator`. This needs to be done because
+Cryomagnetics Instruments are special.
+"""
+
+# IMPORTS #####################################################################
+from __future__ import absolute_import
+
+from serial.tools.list_ports import comports
+from serial import SerialException
+from instruments.abstract_instruments.comm import (
+    CryomagneticsSerialCommunicator, serial_manager
+)
+
+from instruments.abstract_instruments.instrument import Instrument
+
+# CLASSES #####################################################################
+
+
+class CryomagneticsInstrument(Instrument):
+    """
+    Base class for Cryomagnetics Instruments
+    """
+
+    def __init__(self, filelike):
+        Instrument.__init__(self, filelike)
+
+    # pylint: disable=too-many-arguments
+    @classmethod
+    def open_serial(cls, port=None, baud=9600, vid=None, pid=None,
```

Comment:
Yeah, never copy pasta large blocks of code. Especially in this case, it would create a code path that is super similar but not easily discoverable.

With moving your echo handling to using the existing ACK handling, I'm pretty sure this all can be removed.